### DISCUSSION TITLE
Roadmap: future of CF for netCDF versus other file formats / data models

### DISCUSSION THREAD

**Discussion by sadielbartholomew** [cf-convention/discuss#359] [Comments and ideas for changing CF]
### Topic for discussion

For the CF Roadmap paper mapped out at the CF Workshop 2024, we want to review any/all of our our wish list, wishful thinking and plans relating to **the netCDF file format**. This can be a home for discussions relating to this topic, with a view to preparing a section for the paper.

**Comment by davidhassell** [comment]
Hello,

The second half (24-25 September) of the [2025 CF workshop](https://cfconventions.org/Meetings/2025-Workshop.html) is dedicated to the CF roadmap, including hackathons dedicated to advancing the road map topics. These hackathons have no pre-determined structure and can be used for anything from further debate to writing an actual proposal - whatever would be most useful the topic at this time.

If there is interest in working on this roadmap topic in this way, please let us know in a comment on this discussion soon, so that we can advertise it on the meeting agenda - this is a good opportunity to possibly bring in fresh ideas from people who may not yet have engaged with the roadmap.

Thanks, on behalf of the workshop committee.

**Comment by sadielbartholomew** [comment]
Hi @davidhassell, as mentioned and agreed (I believe) in the workshop organisation meeting yesterday, this topic lends itself well to an open discussion so I propose we include a time slot of at least 15 minutes where as many attendees as possible gather and discuss the topic (so, ideally a non-parallel session). Unless someone else is keen to drive it, I am happy to chair/lead. Ideally we can have more like 30-45 minutes if the schedule allows since this is a very important topic for planning for the future, hence for the roadmap.

My suggestions for questions on the theme to initiate and shape the discussion are:

* How format agnostic is CF, logically and practically? 
* What, if anything, are the barriers to use of CF for other file/data formats?
* What other (to netCDF) file formats and data models are in use with similar application/scope, and can CF cater for these (if so, how much change/effort could it need from both the CF side and the data side)?
* What do we anticipate the future of netCDF to be, both shorter and longer term - do we expect it to be prominent for signfiicantly long or that we may need to start overseeing use of CF for other formats sooner rather than later to ensure longevity of the standard?

**Reply by ChrisBarker-NOAA** [reply]
Sounds good -- important topic indeed.

One thought -- perhaps (or not) it would be helpful to classify other file formats in terms of their compatibility with the netcdf data model. e.g. Zarr, I believe, is fully compatible. That is, applying CF to zarr format storage is trivial.

Not so much with e.g. GRIB.

**Reply by sadielbartholomew** [reply]
Thanks Chris, very good point! If we do get a session on this as we propose here, then that can be a subpoint on the third question in the list above - maybe I can reorder the list so the third comes before the second since your point illustrates that some cases are indeed trivial (at least, on paper, maybe not practically?) so the opposite of a barrier. Or somehow change the discussion prompts to account for your point otherwise.

**Reply by cofinoa** [reply]
FYI, 
zarr-developers/geozarr-spec#84

**Reply by larsbarring** [reply]
Another issue that possibly might be relevant to touch upon in this context: https://github.com/orgs/cf-convention/discussions/377

**Reply by sadielbartholomew** [reply]
Thanks Antonio and Lars, those are indeed very relevant. I'll provide a brief summary of each (and anything else relevant posted on this thread or otherwise) at the start of the session as specific file format concerns (Zarr is obviously a key one to consider).
